modules/ml/src/main/java/org/apache/ignite/ml/composition/combinators/sequential/TrainersSequentialComposition.java

```diff
@@ -0,0 +1,125 @@
+ * @param <O1> Type of output of model produced by first trainer.
+ * @param <O2> Type of output of model produced by second trainer.
+ * @param <L> Type of labels.
+ */
+public class TrainersSequentialComposition<I, O1, O2, L> extends DatasetTrainer<ModelsSequentialComposition<I, O1, O2>, L> {
+    /** First trainer. */
+    private DatasetTrainer<IgniteModel<I, O1>, L> tr1;
+
+    /** Second trainer. */
+    private DatasetTrainer<IgniteModel<O1, O2>, L> tr2;
+
+    /** Dataset mapping. */
+    private IgniteFunction<? super IgniteModel<I, O1>, DatasetMapping<L, L>> datasetMapping;
+
+    /**
+     * Construct sequential composition of given two trainers.
+     *
+     * @param tr1 First trainer.
+     * @param tr2 Second trainer.
+     * @param datasetMapping Dataset mapping.
+     */
+    public TrainersSequentialComposition(DatasetTrainer<? extends IgniteModel<I, O1>, L> tr1,
+        DatasetTrainer<? extends IgniteModel<O1, O2>, L> tr2,
+        IgniteFunction<? super IgniteModel<I, O1>, DatasetMapping<L, L>> datasetMapping) {
+        this.tr1 = CompositionUtils.unsafeCoerce(tr1);
+        this.tr2 = CompositionUtils.unsafeCoerce(tr2);
+        this.datasetMapping = datasetMapping;
+    }
+
+    /** {@inheritDoc} */
+    @Override public <K, V> ModelsSequentialComposition<I, O1, O2> fit(DatasetBuilder<K, V> datasetBuilder,
+        IgniteBiFunction<K, V, Vector> featureExtractor, IgniteBiFunction<K, V, L> lbExtractor) {
+
+        IgniteModel<I, O1> mdl1 = tr1.fit(datasetBuilder, featureExtractor, lbExtractor);
+        DatasetMapping<L, L> mapping = datasetMapping.apply(mdl1);
+
+        IgniteModel<O1, O2> mdl2 = tr2.fit(datasetBuilder,
+            featureExtractor.andThen(mapping::mapFeatures),
+            lbExtractor.andThen(mapping::mapLabels));
+
+        return new ModelsSequentialComposition<>(mdl1, mdl2);
+    }
+
+    /** {@inheritDoc} */
+    @Override public <K, V> ModelsSequentialComposition<I, O1, O2> update(
+        ModelsSequentialComposition<I, O1, O2> mdl, DatasetBuilder<K, V> datasetBuilder,
+        IgniteBiFunction<K, V, Vector> featureExtractor, IgniteBiFunction<K, V, L> lbExtractor) {
+
+        IgniteModel<I, O1> firstUpdated = tr1.update(mdl.firstModel(), datasetBuilder, featureExtractor, lbExtractor);
+        DatasetMapping<L, L> mapping = datasetMapping.apply(firstUpdated);
+
+        IgniteModel<O1, O2> secondUpdated = tr2.update(mdl.secondModel(),
+            datasetBuilder,
+            featureExtractor.andThen(mapping::mapFeatures),
+            lbExtractor.andThen(mapping::mapLabels));
+
+        return new ModelsSequentialComposition<>(firstUpdated, secondUpdated);
+    }
+
+    /** {@inheritDoc} */
+    @Override protected boolean checkState(ModelsSequentialComposition<I, O1, O2> mdl) {
+        // Never called.
+        throw new IllegalStateException();
+    }
+
+    /** {@inheritDoc} */
+    @Override protected <K, V> ModelsSequentialComposition<I, O1, O2> updateModel(
```

Comment:
Update javadoc to reflect the fact that this method shell not be called.

Reply:
For the moment, no.